test/test_volatility.py

```diff
@@ -0,0 +1,381 @@
+    args = globalargs + [
+        "--single-location",
+        img,
+        "-q",
+        plugin,
+    ] + pluginargs
+
+    return runvol(args, volatility, python)
+
+#
+# TESTS
+#
+
+# WINDOWS
+
+def test_windows_pslist(image, volatility, python):
+    rc, out, err = runvol_plugin("windows.pslist.PsList", image, volatility, python)
+    out = out.lower()
+    assert out.find(b"system") != -1
+    assert out.find(b"csrss.exe") != -1
+    assert out.find(b"svchost.exe") != -1
+    assert out.count(b"\n") > 10
+    assert rc == 0
+    assert rc == 0
+
+    rc, out, err = runvol_plugin(
+        "windows.pslist.PsList", image, volatility, python, pluginargs=["--pid", "4"])
+    out = out.lower()
+    assert out.find(b"system") != -1
+    assert out.count(b"\n") < 10
+    assert rc == 0
+    assert rc == 0
+
+def test_windows_psscan(image, volatility, python):
+    rc, out, err = runvol_plugin("windows.psscan.PsScan", image, volatility, python)
+    out = out.lower()
+    assert out.find(b"system") != -1
+    assert out.find(b"csrss.exe") != -1
+    assert out.find(b"svchost.exe") != -1
+    assert out.count(b"\n") > 10
+    assert rc == 0
+    assert rc == 0
+
+def test_windows_dlllist(image, volatility, python):
+    rc, out, err = runvol_plugin("windows.dlllist.DllList", image, volatility, python)
+    out = out.lower()
+    assert out.count(b"\n") > 10
+    assert rc == 0
+    assert rc == 0
+
+def test_windows_modules(image, volatility, python):
+    rc, out, err = runvol_plugin("windows.modules.Modules", image, volatility, python)
+    out = out.lower()
+    assert out.count(b"\n") > 10
+    assert rc == 0
+    assert rc == 0
+
+def test_windows_hivelist(image, volatility, python):
+    rc, out, err = runvol_plugin("windows.registry.hivelist.HiveList", image, volatility, python)
+    out = out.lower()
+
+    not_xp = out.find(b"\\systemroot\\system32\\config\\software")
+    if not_xp == -1:
+        assert out.find(b"\\device\\harddiskvolume1\\windows\\system32\\config\\software") != -1
+
+    assert out.count(b"\n") > 10
+    assert rc == 0
+
+def test_windows_dumpfiles(image, volatility, python):
+
+    json_file = open('./test/known_files.json')
+
+    known_files = json.load(json_file)
+
+    failed_chksms = 0
+
+    if sys.platform == 'win32':
+        file_name = ntpath.basename(image)
+    else:
+        file_name = os.path.basename(image)
+
+    try:
+        for addr in known_files["windows_dumpfiles"][file_name]:
+
+            path = tempfile.mkdtemp()
+
+            rc, out, err = runvol_plugin("windows.dumpfiles.DumpFiles", image, volatility, python, globalargs=["-o", path], pluginargs=["--virtaddr", addr])
+
+            for file in os.listdir(path):
+                fp = open(os.path.join(path, file), "rb")
+                if hashlib.md5(fp.read()).hexdigest() not in known_files["windows_dumpfiles"][file_name][addr]:
+                    failed_chksms += 1
+                fp.close()
```

Comment:
`with` syntax does not require a separate close.
```suggestion
                with open(os.path.join(path, file), "rb") as fp:
                    if hashlib.md5(fp.read()).hexdigest() not in known_files["windows_dumpfiles"][file_name][addr]:
                        failed_chksms += 1
```

Reply:
Fixed in https://github.com/volatilityfoundation/volatility3/pull/789/commits/f96f004b13c15b3c0334f7ca96b2c0459a8f9cf1